github/resource_github_organization_ruleset.go

```diff
@@ -659,46 +662,46 @@ func resourceGithubOrganizationRulesetRead(d *schema.ResourceData, meta any) err
 	return nil
 }
 
-func resourceGithubOrganizationRulesetUpdate(d *schema.ResourceData, meta any) error {
+func resourceGithubOrganizationRulesetUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*Owner).v3client
 
 	rulesetReq := resourceGithubRulesetObject(d, meta.(*Owner).name)
 
 	org := meta.(*Owner).name
 	rulesetID, err := strconv.ParseInt(d.Id(), 10, 64)
 	if err != nil {
-		return unconvertibleIdErr(d.Id(), err)
+		return diag.FromErr(unconvertibleIdErr(d.Id(), err))
 	}
 
-	ctx := context.WithValue(context.Background(), ctxId, rulesetID)
+	ctx = context.WithValue(ctx, ctxId, rulesetID)
 
 	var ruleset *github.RepositoryRuleset
 
 	ruleset, _, err = client.Organizations.UpdateRepositoryRuleset(ctx, org, rulesetID, *rulesetReq)
```

Comment:
Checking if the bypass actors have changed and the length is `0` here would allow you to use the clear function to resolve #2952. See https://github.com/integrations/terraform-provider-github/pull/2891#discussion_r2591993478.

Reply:
@stevehipwell The problem here is, that go-github doesn't have `UpdateRulesetClearBypassActor` for Organization rulesets :/